ppr-ui/src/components/collateral/generalCollateral/GenColSummary.vue

```diff
@@ -0,0 +1,118 @@
+        <p class="ma-0">
+          <span v-if="showingHistory">Hide </span>
+          <span v-else>View </span>
+          General Collateral and Ammendments ({{ generalCollateral.length }})
+        </p>
+      </v-btn>
+      <div v-if="showingHistory" class="general-collateral-summary">
+        <v-row v-for="(item, index) in generalCollateral" :key="index" no-gutters>
+          <v-col :class="[{ 'border-btm': index !== baseGenCollateralIndex }, 'pt-10']">
+            <b v-if="item.addedDateTime">{{ asOfDateTime(item.addedDateTime) }}</b>
+            <div v-if="item.descriptionDelete" class="gc-description-delete pt-30px">
+              <v-chip class="badge-delete" color="primary" label text-color="white" x-small>
+                <b>DELETED</b>
+              </v-chip>
+              <p class="pt-5 ma-0">
+                {{ item.descriptionDelete }}
+              </p>
+            </div>
+            <div v-if="item.descriptionAdd" class="gc-description-add pt-30px">
+              <v-chip color="primary" label text-color="white" x-small>
+                <b>ADDED</b>
+              </v-chip>
+              <p class="pt-5 ma-0">
+                {{ item.descriptionAdd }}
+              </p>
+            </div>
+            <div v-if="item.description && index === baseGenCollateralIndex" class="gc-description">
+              <b v-if="registrationFlowType !== RegistrationFlowType.NEW" class="pt-30px">
+                Base Registration Collateral:
+              </b>
+              <p v-if="item.description" class="pt-30px ma-0">
+                {{ item.description }}
+              </p>
+            </div>
+          </v-col>
+        </v-row>
+      </div>
+    </div>
+    <div v-else class="general-collateral-summary pt-10">
+      <p v-if="generalCollateral.length > 0" class="ma-0">
+        {{ generalCollateral[0].description }}
+      </p>
+    </div>
+  </v-container>
+</template>
+<script lang="ts">
+import {
+  defineComponent,
+  reactive,
+  toRefs,
+  computed
+} from '@vue/composition-api'
+import { useGetters } from 'vuex-composition-helpers'
+// local
+import { RegistrationFlowType } from '@/enums' // eslint-disable-line no-unused-vars
+import { GeneralCollateralIF } from '@/interfaces' // eslint-disable-line no-unused-vars
+import { convertDate } from '@/utils'
+
+export default defineComponent({
+  name: 'GenColSummary',
+  setup () {
+    const {
+      getGeneralCollateral,
+      getRegistrationFlowType
+    } = useGetters<any>(['getGeneralCollateral', 'getRegistrationFlowType'])
+
+    const localState = reactive({
+      showingHistory: false,
+      baseGenCollateralIndex: computed(() => {
+        return (localState.generalCollateral?.length || 0) - 1
+      }),
+      generalCollateral: computed((): GeneralCollateralIF[] => {
+        return getGeneralCollateral.value as GeneralCollateralIF[] || []
+      }),
+      registrationFlowType: computed((): RegistrationFlowType => {
```

Comment:
nit: but I don't think we need this computed in the localState...
just const registrationFlowType = getRegistrationFlowType.value 
and then return it below

Reply:
for now that would be okay since we only use the summary in a situation where the flow is static, but doing it that way would prevent it from updating if it was ever in a situation where it changed after it first rendered

Reply:
Since this will be re-used in the future I think it's better to keep it the way it is unless there's a reason this is worse?